### PR TITLE
Precise Rotation

### LINKED ISSUE #476: Direct Text Input to < .3 for SS values (don't synch when less than .3)
As some headsets i.e. pimax users might have settings like 2x in pi-tool, and .25 in SVR.... we shouldn't adjust SS values that are < .3

### LINKED ISSUE #481: More accurate HMD Rotations tracking
**Is your feature request related to a problem? Please describe.**
The HMD Rotations value can drift drastically over a long session due to accumulated error. This makes the center marker rotation arrows near pointless. For example I had a 3 1/2 hour play session end up saying I was rotated an extra 2.7 turns CCW

**Describe the solution you'd like**
More accurate/absolute rotation tracking of HMD Rotations from the positional data

**Describe alternatives you've considered**
Obvious one is just manually resetting the rotation value often during play. 
Maybe if absolute rotation tracking isn't possible, implement a feature that auto resets the HMD Rotations value when it detects that the HMD is facing the same direction as when the counter initialized and turns value is < .5 

**Additional context**
We discussed this on the Discord and was told that is should be technically possible to get near 100% accurate, just it wasn't trivial

### LINKED ISSUE #488: Playspace Moves Away From Me
**Describe the bug**
Today july 9th when I use playspace, my playspace moves away from me. 

There was a steam vr update today "SteamVR 1.13 makes it convenient to re-center and re-orient your play area across all games. Check out the new button in the dashboard and try it for yourself."

**To Reproduce**
1. Use playspace mover

**Expected behavior**
I expect it to work like it's always has. If i enable the old "use frame update playspace mover setting" it works as expected but is very unoptimized and lags.

**Screenshots**
test bug (2)

**Software (please complete the following information):**
 - OS: Windows 10
 - Advanced Settings Version: Tried both 4.0.1 and 5.2

**Hardware (please complete the following information):**
 - CPU: i9-9900k
 - Graphics Card: 2080ti RTX 
 - HMD: Vive Pro

**Additional context**
I restarted my computer, reinstalled 2 version of advanced settings, and re-settup my room settings. I'll look into reverting the steamvr version to see if this fixes it. If it does i'll let you know.

### PR DESCRIPTION
**Pull request by ykeara**
fixes #476
fixes #481 
fixes #488

### Moved Stuff

- Adjust Chaperone Is now Defaulted to Off, as SteamVR changed chaperone Behavior in 1.13, This can still be re-enabled In settings. [moved from offsets tab]

### Bug Fixes

- Fixes some errors with super sampling, forcing SuperSample values to 1 digit of precision, should be able to arbitrarily set now

- Changes how We count Rotations.

- Fixed an Issue where the scroll box would cover other elements in the settings page.

### Rotation Counter Change

- New method uses absolute position to prevent accumulation of error
- New Method may fail if a user  180 deg / 7 frame interval (Worst Case scenario of 30 fps this is ~770 deg/s) [we do nothing, but would realistically skip a rotation if hit, but I would think this would be more likely caused by a rouge IMU than a human hitting this point (and a Quest 72 hz at full this is ~1800 deg/sec turning speeds)
- New Method Changes 0 point to forward of playspace rather than arbitrary, This is easy to switch, but requires more calc time to have arbitary point (another quaternion multiplication) I am un-sure what makes the most sense

- Because 0 point is now forward, this means if you have rotation @ .75 rotaitons CCW, and Reset you will be at .25 rotations CW after reset. [and if you have 1.75 CCW it will be .25 CW after reset]

- Turns are ignored when the HMD is inverted (roll >90 deg L/R or pitch >  or < straight vertical)

### CODE REVIEW

File: src/tabcontrollers/MoveCenterTabController.cpp [hunk 1]
```diff
@@ -2144,48 +2144,42 @@ void MoveCenterTabController::updateHmdRotationCounter(
         m_lastHmdQuaternion = m_hmdQuaternion;
         return;
     }
-    // Construct a quaternion representing difference between old
-    // hmd pose and new hmd pose.
-    vr::HmdQuaternion_t hmdDiffQuaternion = quaternion::multiply(
-        m_hmdQuaternion, quaternion::conjugate( m_lastHmdQuaternion ) );
```
[the remark this answers is not in the document]
Comment (ykeara) [reply src/tabcontrollers/MoveCenterTabController.cpp:2146]: As Discussed you are right, the  `north` and `south` pole's would be critical regions where this method will more or less fail.

will move forward w/ a proposed hybrid system, synching occasionally, and using results based on Pitch/roll

[the remark this answers is not in the document]
Comment (ykeara) [reply src/tabcontrollers/MoveCenterTabController.cpp:2146]: Soo The failure condition it appears is at the 180 deg mark (in xz plane) if you are inverted (i.e. hmd is "upside down" normal looking should be considered 90 deg... so > 90 deg change in either vertical direction will add a count of the opposite direction.

The Easiest solution being to add an inversion check to just add the opposite counter if you are inverted might make the most sense.


Between the (imo) odd representation of angles that quaternion (get yaw) generates I think this is all we should do as well because an inversion of hmd doesn't In almost all cases mean an actual turn event........ (maybe some weird side-ways flip would be.... edge of edge of edge)

tl;dr I think a simple check of pitch & roll, and change the increment counter to the opposite if HMD is inverted  is all we should really do.

I believe this should cover everything aside from things like side-ways flips and spinning on top of your head.... which quite frankly I think is okay if we leave it off XD.

perhaps we could add a logging error on inversion though again if somebody is sleeping and their head is at > 90 deg up (unlikely for natural posture) it would get spammy in logs.

**Comment by ykeara** [comment]
Now Ignores rotation counts when HMD is inverted (should be noted that rolling your head through the critical area can still generate a "turn" (as technically the headset did turn) This can probably be improved upon in the future, but we need to think about just exactly what we want the behavior to be is (as I think its impossible to program for all edge cases in this instance.... as hmd rotation is not body rotation.

**Comment by Kung-vr** [comment]
Yeah since you've noted that master build rotation also fails when inverted this PR is better to have merged rather than wait. Long term might be interesting to go research some math about rope twisting connected to rigid objects because the wire twist in the ground-truth that we want to preserve. Would potentially be a huge pain though to reconcile something like that in a non-relative or in other words absolute way with the tracking coordinates though.
